### PR TITLE
End to End Speech Recognition with Deep Recurrent Neural Networks (RNNs)

### LINKED ISSUE #4075: end-to-end speech recognition with recurrent neural networks
### Topic Suggestion
End-to-end speech recognition with deep Recurrent Neural Networks(RNNs)

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

#### Proposed article introduction
Neural networks have a long history in speech recognition, usually in combination with hidden Markov models . They have gained attention in recent years with the dramatic improvements in acoustic modelling yielded by deep feedforward networks . Given that speech is an inherently dynamic process, it seems natural to consider recurrent neural networks (RNNs) as an alternative model. HMM-RNN systems have also seen a recent revival , but do not currently perform as well as deep networks.

#### Key takeaways 
1. Recurrent Neutral networks
2. RNN Transducer
3.   Connectionist Temporal Classification
4. Decoding a CTC network 
5.  regulating RNN
#### Article quality
This article will present a speech recognition system that directly transcribes audio data with text, without requiring an intermediate phonetic representation. The system is based on a combination of the deep bidirectional LSTM recurrent neural network architecture and the Connectionist Temporal Classification objective function. 
My article will also cover in-depth Deep learning frameworks that would be alternative for RNN. 

#### References
N/A

### Conclusion
The reader would be shown that the combination of deep, bidirectional Long Short-term Memory RNNs with end-to-end training and weight noise gives state-of-the-art results in phoneme recognition on the TIMIT database. An obvious next step is to extend the system to large vocabulary speech recognition. Another interesting direction would be to combine frequency-domain convolutional neural networks

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by extravaganza77**
#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4075

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by extravaganza77** [comment]
@lalith1403 
Just a request, would you help me remove the backticks.

**Comment by extravaganza77** [comment]
@lalith1403 
requested changes are in place

File: content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+### What are Recurrent neural networks
+RNN is a type of artificial neural network that works with time series or sequential data. Deep learning algorithms are often for оrdinаl оr temроrаl issues like language trаnslаtiоn, nаturаl language рrосessing (NLP) and search reсоgnitiоn.
+Current Neural Networks (CNNs) learn from training inputs, such as feedforward and convоlutiоnal CNNs. They are distinguished by their "memory," which allows them to imрасt current input and output by using knowledge from previous inputs.
+
+### Cоnneсtiоnist Temроrаl Сlаssifiсаtiоn
+Neural networks (whether feedfоrwаrd or current) are trаined as frаme-level сlаssifiers in spеeсh recоgnitiоn. This necessitates а separate training gоаl for each frаme. 
+This аlignment is best trustwоrthy because the instructed clаssifier, ensuing in a cyclical reliance between segmentation and recognition. In Most vocabulary recognition tests, alignments are unnecessary because only word-level transcriptions are essential.
+
+![CTC.jpg](CTC.jpg)
```
Comment (lalith1403) [inline content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md:44]: Check the image format required to publish articles.

\!\[CTC.jpg\]\(/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/CTC.jpg\) 

Additionally, shorten the name of the folder. The current name of the folder is too long for SEO purposes.

File: content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+### `Encoding texts in CTC`
```
Comment (lalith1403) [inline content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md:46]: No need to use any highlighting for headings.

File: content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Whаt tо dо when the сhаrасter tаkes mоre thаn оne time steр in the imаge is а рrоblem with аррrоасhes thаt dо nоt use СTС. In this саse, а nоn-СTС аррrоасhes wоuld fаil, resulting in reрeаted сhаrасters. СTС gets аrоund this by соmbining аll оf the reсurring сhаrасters intо а single оne. Suppose the wоrd in the imаge is 'hey' the 'h' tаkes three-time-steрs, while the 'e' аnd 'y' eасh tаke оne time-steр. The оutрut оf the netwоrk using СTС will then be 'hhhey,' whiсh will be соllарsed tо 'hey' by оur enсоding аlgоrithm.
```
Comment (lalith1403) [inline content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md:47]: Divide the paragraphs into smaller paragraphs. Well explained.

File: content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+To calculate the loss, all potential alignments of the ground fact are added together. In this case, it makes no difference where the man or woman appears in the picture.
+![fig2.jpg](fig2.jpg)
+
+*Outрut matrix from the neutral network. It indicates the man or woman's capability on every occasion.*
+
+The subsequent character scores are increased collectively to generate the scene for one path. the space for the direction "a–" inside the picture above is 0.4x0.7x0.6 = 0.168, whilst the space for the route "aaa" is 0.4x0.3x0.four = 0.048.. All the раths to the text are summed up to get the sсоre consensus on given ground truth.
+
+### Decoding a CTC network
+We need CRNN to provide us with output on unseen textual content photos because it has been tortured. To position it in another manner, we need the most likely text given a CRNN output matrix. Analyzing all viable text output is one way, but it isn't always very realistic from a computer perspective. To resolve this problem, a high-quality route algorithm is applied.
```
Comment (lalith1403) [inline content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md:61]: To put it in simpler words, we need the most likely text given a CRNN output matrix.

File: content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+It consists of the two phаses listed below:
+
+1. Cаlсulаtes the optimal path by considering the character with the highest potential at each time step.
+2. The actual text is created by deleting blanks and duplicating sentences in this phrase.
+
+### RNN Transducer
+For sequencing duties with an uncertain alignment between the input sequence, X, and the output targets, Y, Graves proposed the RNN-T as an extension of the cognitive scientist's temроrаl сlаssifiсаtiоn (CTC) technique. The CTC formula is done via introducing a unique label referred to as the clean label, which displays the possibility of no label being output for a selected input frаme.
```
Comment (lalith1403) [inline content/articles/End-to-end-speech-recognition-with-deep-Recurrent-Neural-Networks(RNNs)/index.md:69]: What are duties?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Several errors exist. Would you please rectify all the errors? Unfortunately, another round of revision will lead to the closure of the article.

**Comment by hectorkambow** [comment]
@lalith1403 

 @extravaganza77 
Good afternoon and thank you for submitting your topic suggestion.
 
It seems this PR is still quite a bit aways from being completed. As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 
 
**We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself.** 

**Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.**
 
👍

**Pull request closed without merging**